### PR TITLE
Add width and height check for MujocoRenderer

### LINKED ISSUE #1225: [Proposal] Add the width and height check of the `OffScreenViewer`
### Proposal

The `width` and `height` of `OffScreenViewer` cannot be None otherwise, it will cause the error. Adding a type check before creating the opengl_context could provide more information for debugging, at least the glfw backend.

### Motivation

I used the `MujocoRenderer` to render a MuJoCo model outside the gymnasium project. And I forgot to set the `width` and `height` because the `gymnasium==0.29.1` has the default value.

```
self.renderer = MujocoRenderer(self.model, self.data, CAMERA_CONFIG)
```

So the following error occurs.
```
Exception ignored in: <function GLContext.__del__ at 0x111d33c70>
Traceback (most recent call last):
  File "envs/vis/lib/python3.10/site-packages/mujoco/glfw/__init__.py", line 45, in __del__
    self.free()
  File "envs/vis/lib/python3.10/site-packages/mujoco/glfw/__init__.py", line 38, in free
    if self._context:
AttributeError: 'GLContext' object has no attribute '_context'
RuntimeError: No OpenGL backend could be imported. Attempting to create a rendering context will result in a RuntimeError.
Exception ignored in: <function OffScreenViewer.__del__ at 0x1101436d0>
Traceback (most recent call last):
  File "envs/vis/lib/python3.10/site-packages/gymnasium/envs/mujoco/mujoco_rendering.py", line 202, in __del__
    self.free()
  File "envs/vis/lib/python3.10/site-packages/gymnasium/envs/mujoco/mujoco_rendering.py", line 199, in free
    self.opengl_context.free()
AttributeError: 'OffScreenViewer' object has no attribute 'opengl_context'
```

The traceback info is not straightforward for debugging, maybe adding a type check is better.

### Pitch

_No response_

### Alternatives

_No response_

### Additional context

_No response_

### Checklist

- [X] I have checked that there is no similar [issue](https://github.com/Farama-Foundation/Gymnasium/issues) in the repo

### PR DESCRIPTION
**Pull request by Beanpow**
# Description

The width and height of OffScreenViewer cannot be None otherwise, it will cause the error. Adding a type check before creating the opengl_context could provide more information for debugging.

Fixes #1225

## Type of change

Please delete options that are not relevant.

- [ ] Documentation only change (no code changed)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# Checklist:

- [x] I have run the [`pre-commit` checks](https://pre-commit.com/) with `pre-commit run --all-files` (see `CONTRIBUTING.md` instructions to set it up)
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

**Comment by Beanpow** [comment]
@Kallinteris-Andreas could you review this code?

File: gymnasium/envs/mujoco/mujoco_rendering.py
Comment (pseudo-rnd-thoughts) [inline gymnasium/envs/mujoco/mujoco_rendering.py:721]: Could you specify what the width and height values are to help users debug
`f'The width ({width}) and height ({height}) cannot be `None`'` ...

**Review by pseudo-rnd-thoughts — CHANGES_REQUESTED** [review]

**Review by pseudo-rnd-thoughts — APPROVED** [review]

**Review by Kallinteris-Andreas — COMMENTED** [review]
will be merged after https://github.com/Farama-Foundation/Gymnasium/pull/1229

**Review by Kallinteris-Andreas — APPROVED** [review]
LGTM I will merge after https://github.com/Farama-Foundation/Gymnasium/pull/1229 , @Beanpow you may need to do a merge then,

**Comment by pseudo-rnd-thoughts** [comment]
@Kallinteris-Andreas are you sure we can't merge now?

**Pull request merged**
